### PR TITLE
Added new files configure.fv3.stellar, modules.fv3.stellar

### PR DESCRIPTION
**Pull request by mrudko**
Dear Colleges,

I have added the files configure.fv3.stellar and modules.fv3.stellar needed for configuring fv3gfs on Stellar cluster. 

Comments:

 - Compiler directives:

   To compile and link the model, Fortran (mpif90) and C (mpicc) mpi wrappers are used. They become available on Stellar cluster after loading the modules from modules.fv3.stellar file. Please issue the command "source modules.fv3.stellar".

- Before submitting job, the environmental variable $LD_LIBRARY_PATH needs to be updated with the paths to the local libraries libraries (ESMF, NCEP, FMS). Please issue the command "export LD_LIBRARY_PATH = $LD_LIBRARY_PATH:/path_to_local_library"

Mike

### CODE REVIEW

File: FV3/conf/configure.fv3.stellar [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+############
+#export NCEPLIBS_DIR=$INSTALL_PREFIX/NCEPlibs
+#export ESMFMKFILE=$INSTALL_PREFIX/ESMFlibs/lib/libO/Linux.intel.64.mpiuni.default/esmf.mk
+#export FMS_DIR=$INSTALL_PREFIX/fv3gfs-fortran/FMS
+
+#Note that the variable $INSTALL_PREFIX needs to be exported prior configuring the environment. An example of the absolute pathes to the libraries is given below.
+
+export NCEPLIBS_DIR=/home/mr7417/NCEPlibs/NCEPlibs
+export ESMFMKFILE=/home/mr7417/ESMFlibs/lib/libO/Linux.intel.64.mpiuni.default/esmf.mk
+export FMS_DIR=/home/mr7417/fv3gfs-fortran/FMS
```
Comment (spencerkclark) [inline FV3/conf/configure.fv3.stellar:12]: nit: I like the intent of this comment; I agree it would good to clarify which variables the configure files expect to be defined.  However, I think we can leave it up to the user how they would like to define those (i.e. we do not need to assume that these libraries are installed within the same `INSTALL_PREFIX`).  I might suggest rewording this to something like the following:
```suggestion
# Note it is expected that the NCEPLIBS_DIR, FMS_DIR, and ESMFMKFILE environment 
# variables are exported before using this configure file.  The NCEPLIBS_DIR and
# FMS_DIR variables refer to the install locations of NCEPlibs and FMS, respectively,
# and the ESMFMKFILE variable refers to the location of the esmf.mk file associated 
# with the installation of ESMF.
```

File: FV3/conf/configure.fv3.stellar [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+need := 3.81
+ok := $(filter $(need),$(firstword $(sort $(MAKE_VERSION) $(need))))
+ifneq ($(need),$(ok))
+$(error Need at least make version $(need).  Load module gmake/3.81)
+endif
+
+NETCDF_ROOT = $(NETCDF_DIR)
+INCLUDE = -I$(NETCDF_ROOT)/include
+
+# The fpp option -D__GFORTRAN__(-Dname) is specified to avoid the unwanted errors when compiling the model. The source of the errors is the mismatch between the way the "intent" of the arguments are defined in the interface definition of the proceduce pointer used in the atmos_model.F90 and the way the "intent" of the arguments are defined in the procedures the pointet points to
```
Comment (spencerkclark) [inline FV3/conf/configure.fv3.stellar:49]: nit: could you split this comment onto multiple lines?  It will make it easier to read in a terminal window.

Comment (spencerkclark) [inline FV3/conf/configure.fv3.stellar:49]: typo:
```suggestion
# The fpp option -D__GFORTRAN__(-Dname) is specified to avoid the unwanted errors when compiling the model. The source of the errors is the mismatch between the way the "intent" of the arguments are defined in the interface definition of the proceduce pointer used in the atmos_model.F90 and the way the "intent" of the arguments are defined in the procedures the pointer points to
```

File: FV3/conf/modules.fv3.stellar [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# NOTE: the "module purge" and loading of the module command are
+# handled by the module-setup.sh (or .csh) script.
+##
+## load programming environment
+## this typically includes compiler, MPI and job scheduler
+##
+
+# Kai's modules (note these are commented out in favor of the more up-to-date versions):
```
Comment (spencerkclark) [inline FV3/conf/modules.fv3.stellar:8]: In formalizing this, I think it would be good to include Kai's full name here:
```suggestion
# Modules suggested by Kai-Yuan Cheng.  Note these are commented out in favor
# of the more up-to-date versions):
```

File: FV3/conf/modules.fv3.stellar [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# module load netcdf/intel-19.1/hdf5-1.10.6/4.7.4
+# module load hdf5/intel-19.1/1.10.6
+
+# Up-to-date equivalents of Kai's modules:
+module load intel/2021.1.2
+module load openmpi/intel-2021.1/4.1.2
+module load netcdf/intel-2021.1/hdf5-1.10.6/4.7.4
+module load hdf5/intel-2021.1/1.10.6
+
+# Kai's environment variables
```
Comment (spencerkclark) [inline FV3/conf/modules.fv3.stellar:20]: Since we typically do not set this variable explicitly in modules files, it would be helpful to add some context for why:
```suggestion
# Per Kai-Yuan Cheng, the netcdf module on Stellar sets the NETCDFDIR variable
# rather than the NETCDF_DIR variable, the variable used on existing supported
# platforms.  For compatibility we therefore add a NETCDF_DIR variable to the
# environment which takes the value of the NETCDFDIR variable.
```

File: FV3/conf/modules.fv3.stellar [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+## this typically includes compiler, MPI and job scheduler
+##
+
+# Kai's modules (note these are commented out in favor of the more up-to-date versions):
+# module load intel/19.1
+# module load openmpi/intel-19.1/4.1.0
+# module load netcdf/intel-19.1/hdf5-1.10.6/4.7.4
+# module load hdf5/intel-19.1/1.10.6
+
+# Up-to-date equivalents of Kai's modules:
```
Comment (spencerkclark) [inline FV3/conf/modules.fv3.stellar:14]:
```suggestion
```

**Review by spencerkclark — COMMENTED** [review]
Thanks @mrudko -- generally looks good.  I added some comments mostly aimed at cleaning things up a little more and making things more compatible with some of the rest of our software.

File: FV3/conf/configure.fv3.stellar [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+HYDRO = N
+32BIT = N
+
+include $(ESMFMKFILE)
+ESMF_INC = $(ESMF_F90COMPILEPATHS)
+ESMF_LIB = $(ESMF_F90ESMFLINKPATHS) $(ESMF_F90ESMFLINKLIBS)
+
+NEMSIOINC = -I$(NCEPLIBS_DIR)/include
+#NCEPLIBS = $(ESMF_LIB) -L$(NCEPLIBS_DIR)/lib -lnemsio_d -lbacio_4 -lsp_v2.0.2_d -lw3emc_d -lw3nco_d
+NCEPLIBS = $(ESMF_LIB) -L$(NCEPLIBS_DIR)/exec_linux.intel -lnemsio_d -lbacio_4 -lsp_v2.0.2_d -lw3emc_d -lw3nco_d
```
[the remark this answers is not in the document]
Comment (mrudko) [reply FV3/conf/configure.fv3.stellar:36]: The install_nceplibs.sh works perfectly fine if one supplies the value of INSTALL_PREFIX in a correct form. In my local home directory the proper usage of the command is 
./install_nceplibs.sh NCEPlibs /home/mr7417/NCEPlibs cheyenne intel. Note that specifying $PLATFORM as cheyenne is necessary in order avoid the errors when installing the NCEP library. The source of the error is the mismatch between Fortran mpi wrappers (mpif90 and mpiifort)

File: FV3/conf/configure.fv3.stellar [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+FFLAGS += $(FFLAGS_OPT)
+FAST := $(TRANSCENDENTALS)
+endif
+
+ifneq ($(OPENMP),)
+CFLAGS += $(CFLAGS_OPENMP)
+FFLAGS += $(FFLAGS_OPENMP)
+LDFLAGS += $(LDFLAGS_OPENMP)
+# to correct a loader bug on gaea: envars below set by module load inteli. Note this line might not be nesseseary for Stellar cluster.
+LIBS += -L$(INTEL_PATH)/$(INTEL_MAJOR_VERSION)/$(INTEL_MINOR_VERSION)/lib/intel64 -lifcoremt
```
[the remark this answers is not in the document]
Comment (mrudko) [reply FV3/conf/configure.fv3.stellar:140]: Compiling the model without this line went well, the code runs perfectly fine, the line has been deleted.

File: FV3/conf/modules.fv3.stellar [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+module load openmpi/intel-2021.1/4.1.2
+module load netcdf/intel-2021.1/hdf5-1.10.6/4.7.4
+module load hdf5/intel-2021.1/1.10.6
+
+# Kai's environment variables
+export NETCDF_DIR=${NETCDFDIR}
+export FC=mpif90
+export CC=mpicc
+export CXX=mpicxx
+export LD=mpif90
```
[the remark this answers is not in the document]
Comment (mrudko) [reply FV3/conf/modules.fv3.stellar:25]: Done

File: FV3/conf/configure.fv3.stellar (hunk 2 above)
Comment (mrudko) [reply FV3/conf/configure.fv3.stellar:49]: Done

Comment (mrudko) [reply FV3/conf/configure.fv3.stellar:49]: Done

File: FV3/conf/configure.fv3.stellar [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+#Note that the variable $INSTALL_PREFIX needs to be exported prior configuring the environment. An example of the absolute pathes to the libraries is given below.
+
+export NCEPLIBS_DIR=/home/mr7417/NCEPlibs/NCEPlibs
+export ESMFMKFILE=/home/mr7417/ESMFlibs/lib/libO/Linux.intel.64.mpiuni.default/esmf.mk
+export FMS_DIR=/home/mr7417/fv3gfs-fortran/FMS
+
+FC = mpif90
+CC = mpicc
+LD = $(FC)
```
[the remark this answers is not in the document]
Comment (mrudko) [reply FV3/conf/configure.fv3.stellar:12]: I added the definition of CXX variable to the configure.fv3.stellar script. However, I do not completely understand what is meant by following the pattern of setting default values? Is this the order of variables in the file or the condition (ifeq ($(origin, FC), default) endif) ?

**Review by spencerkclark — APPROVED** [review]
Perfect, thanks @mrudko.  I noticed a couple more minor things to clean up after merging in mrudko#1.  If you'd like I think you can use the GitHub interface to commit the suggestions and then this should be good to go.  I'll look more into getting the test suite to run later today.

**Comment by spencerkclark** [comment]
Thanks @mrudko!
